Gems/ROS2/Code/Source/ProximitySensor/ROS2ProximitySensor.cpp

```diff
@@ -0,0 +1,144 @@
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<ROS2ProximitySensor, ROS2SensorComponent>()->Version(1)->Field(
+                "detectionDistance", &ROS2ProximitySensor::m_detectionDistance);
+
+            if (AZ::EditContext* editContext = serialize->GetEditContext())
+            {
+                editContext->Class<ROS2ProximitySensor>("ROS2 Proximity Sensor", "Proximity object detection component")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &ROS2ProximitySensor::m_detectionDistance,
+                        "Detection distance",
+                        "The maximum distance from where object is detected")
+                    ->Attribute(AZ::Edit::Attributes::Min, 0.);
+            }
+        }
+    }
+
+    ROS2ProximitySensor::ROS2ProximitySensor()
+    {
+        TopicConfiguration tc;
+        AZStd::string type = "std_msgs::msg::bool";
+        tc.m_type = type;
+        tc.m_topic = "proximity_sensor";
+        m_sensorConfiguration.m_frequency = 10;
+        m_sensorConfiguration.m_publishersConfigurations.insert(AZStd::make_pair(type, tc));
+    }
+
+    void ROS2ProximitySensor::Activate()
+    {
+        auto ros2Node = ROS2Interface::Get()->GetNode();
+        AZ_Assert(m_sensorConfiguration.m_publishersConfigurations.size() == 1, "Invalid configuration of publishers for proximity sensor");
+
+        const TopicConfiguration& publisherConfig = m_sensorConfiguration.m_publishersConfigurations["std_msgs::msg::bool"];
+        AZStd::string fullTopic = ROS2Names::GetNamespacedName(GetNamespace(), publisherConfig.m_topic);
+        m_detectionPublisher = ros2Node->create_publisher<std_msgs::msg::Bool>(fullTopic.data(), publisherConfig.GetQoS());
+
+        if (m_sensorConfiguration.m_visualise)
+        {
+            auto* entityScene = AZ::RPI::Scene::GetSceneForEntityId(GetEntityId());
+            m_drawQueue = AZ::RPI::AuxGeomFeatureProcessorInterface::GetDrawQueueForScene(entityScene);
+        }
+
+        AZ::Transform entityTransform;
+        AZ::TransformBus::EventResult(entityTransform, GetEntityId(), &AZ::TransformBus::Events::GetWorldTM);
+
+        m_detectionDirection = AZ::Vector3::CreateAxisX();
+        m_detectionDirection = entityTransform.TransformVector(m_detectionDirection);
+        m_detectionDirection.Normalize();
+
+        ROS2SensorComponent::Activate();
+    }
+
+    void ROS2ProximitySensor::Deactivate()
+    {
+        m_detectionPublisher.reset();
+        ROS2SensorComponent::Deactivate();
+    }
+
+    void ROS2ProximitySensor::Visualise()
```

Comment:
nit: In general, us_en spelling is preferred
```suggestion
    void ROS2ProximitySensor::Visualize()
```

Reply:
This spelling mistake is inherited from ROS2SensorComponent and other parts of the code have the same error. This will be addressed in a separate PR.